tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+  
+  # generate contraction order if necessary
+  flat_connect = np.concatenate(connect_list)
+  if cont_order is None:
+    cont_order = np.unique(flat_connect[flat_connect > 0])
+  else:
+    cont_order = np.array(cont_order)
+  
+  # do all partial traces
+  for ele in range(num_tensors):
+    uni_inds, counts = np.unique(connect_list[ele],return_counts=True)
+    tr_inds = uni_inds[counts==2]
+    temp_connect = connect_list[ele]
+    temp_dims = tensor_dims[ele]
+    for ind in tr_inds:
+      temp_dims = temp_dims[temp_connect != ind]
+      temp_connect = temp_connect[temp_connect != ind]
+      cont_order = np.delete(cont_order, np.flatnonzero(cont_order == ind))
+      
+    tensor_dims[ele] = temp_dims
+    connect_list[ele] = temp_connect
+  
+  # do all binary contractions
+  while len(cont_order) > 0:
+    # identify tensors to be contracted
+    cont_ind = cont_order[0]
+    locs = [ele for ele in range(len(connect_list)) 
+            if sum(connect_list[ele] == cont_ind) > 0]
+  
+    # do binary contraction
+    cont_many, A_cont, B_cont = np.intersect1d(connect_list[locs[0]], 
+                                               connect_list[locs[1]], 
+                                               assume_unique=True, 
+                                               return_indices=True)
+    A_dim = np.sum(tensor_dims[locs[0]])
+    B_dim = np.sum(tensor_dims[locs[1]]) 
+    cont_dim = np.sum((tensor_dims[locs[0]])[A_cont])
+    single_cost = A_dim + B_dim - cont_dim
+    if total_cost is None:
+      total_cost =  single_cost
+    else:
+      total_cost =  total_cost + np.log10(1+10**(single_cost-total_cost))
+    
+    tensor_dims.append(np.append(np.delete(tensor_dims[locs[0]], A_cont), 
+                                 np.delete(tensor_dims[locs[1]], B_cont)))
+    connect_list.append(np.append(np.delete(connect_list[locs[0]], A_cont), 
+                                  np.delete(connect_list[locs[1]], B_cont)))
+  
+    # remove contracted tensors from list and update cont_order
+    del tensor_dims[locs[1]]
+    del tensor_dims[locs[0]]
+    del connect_list[locs[1]]
+    del connect_list[locs[0]]
+    cont_order = np.delete(cont_order,np.intersect1d(cont_order,cont_many, 
+                                                     assume_unique=True, 
+                                                     return_indices=True)[1])
+  
+  # do all outer products
+  num_tensors = len(tensor_dims)
+  if (num_tensors > 1):
+    tensor_sizes = np.sort([np.sum(tensor_dims[ele]) 
+                            for ele in range(num_tensors)])
+    for i in range(num_tensors-1):
+      single_cost = tensor_sizes[0] + tensor_sizes[1]
+      tensor_sizes[0] += tensor_sizes[1]
+      tensor_sizes = np.sort(np.delete(tensor_sizes,1))
+      if total_cost is None:
+        total_cost =  single_cost
+      else:
+        total_cost =  total_cost + np.log10(1+10**(single_cost-total_cost))
+      
+  return total_cost
```

Comment:
make sure this is always a number

Reply:
See previous comment. Maybe for the case of a single input tensor we could return a large negative number, e.g. `total_cost=-100`, such that `10**total_cost` was approximately equal to zero as desired?